nixos/modules/services/networking/sozu.nix

```diff
@@ -0,0 +1,479 @@
+      example = {
+        listeners = [
+          # Example for an HTTP (plaintext) listener
+          {
+            # Possible values are http, https or tcp
+            protocol = "http";
+            # Listening address
+            address = "127.0.0.1:8080";
+
+            # Specify a different IP than the one the socket sees, for logs and
+            # forwarded headers
+            public_address = "1.2.3.4:80";
+
+            # Configures the client socket to receive a PROXY protocol header
+            #expect_proxy = false;
+
+            # Path to custom 404 and 503 answers
+            # A 404 response is sent when sozu does not know about the requested domain or path
+            # A 503 response is sent if there are no backend servers available
+            answer_404 = "../lib/assets/404.html";
+            answer_503 = "../lib/assets/503.html";
+
+            # Defines the sticky session cookie's name, if `sticky_session` is
+            # activated format an application. Defaults to "SOZUBALANCEID"
+            sticky_name = "SOZUBALANCEID";
+          }
+
+          # Example for an HTTPS (OpenSSL or rustls based) listener
+          {
+            # Possible values are http, https or tcp
+            protocol = "https";
+            # Listening address
+            address = "127.0.0.1:8443";
+
+            # Specify a different IP than the one the socket sees, for logs and
+            # forwarded headers
+            #public_address = "1.2.3.4:81";
+
+            # Configures the client socket to receive a PROXY protocol header
+            # This option is incompatible with public_addresss
+            #expect_proxy = false;
+
+            # Path to custom 404 and 503 answers
+            # A 404 response is sent when sozu does not know about the
+            # requested domain or path
+            # A 503 response is sent if there are no backend servers available
+            answer_404 = "../lib/assets/404.html";
+            answer_503 = "../lib/assets/503.html";
+
+            # Defines the sticky session cookie's name, if `sticky_session` is
+            # activated format an application. Defaults to "SOZUBALANCEID"
+            sticky_name = "SOZUBALANCEID";
+
+            # Supported TLS versions. Possible values are "SSLv2", "SSLv3",
+            # "TLSv1", "TLSv1.1", "TLSv1.2", "TLSv1.3". Defaults to "TLSv1.2"
+            tls_versions = [ "TLSv1.2" ];
+
+            # Option specific to Openssl based HTTPS listeners
+            #cipher_list = "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256";
+
+            # Option specific to rustls based HTTPS listeners
+            rustls_cipher_list = [ "TLS13_CHACHA20_POLY1305_SHA256" ];
+          }
+        ];
+
+        applications = {
```

Comment:
I'm not really sure about this, especially since I've never used `sozu`, but I feel like there might be value in creating a NixOS option for `applications`. Maybe you benefit from some automatic certificate integration with our `security.acme` module, maybe it makes it easier to provide examples... I'm not sure yet. I'll give this some more thought.

Having a single `settings` option might feel a bit overwhelming to do entirely in `nix` instead of `toml` for some people, if they're coming from a non NixOS distro as an existing `sozu` user.

Reply:
I'll probably need help with this, as I don't know how to go about integrating `security.acme`.